packages/core/src/App/Containers/Layout/app-contents.jsx

```diff
@@ -32,28 +31,34 @@ const AppContents = ({
     const [show_cookie_banner, setShowCookieBanner] = React.useState(false);
     const [is_gtm_tracking, setIsGtmTracking] = React.useState(false);
     const { is_appstore } = React.useContext(PlatformContext);
-    const {
-        client: { user_id },
-    } = useStore();
 
     const tracking_status = tracking_status_cookie.get(TRACKING_STATUS_KEY);
 
     const scroll_ref = React.useRef(null);
 
+    const current_page = window.location.hostname + window.location.pathname;
+
     React.useEffect(() => {
         // rudderstack page view trigger
-        WS.wait('authorize').then(() => {
-            RudderStack.identifyEvent(user_id, {
-                language: getLanguage().toLowerCase() || 'en',
-            });
-            const current_page = window.location.hostname + window.location.pathname;
-            RudderStack.pageView(current_page);
+        WS.wait('authorize').then(response => {
+            if (!response.error) {
+                const { user_id } = response.authorize;
+
+                if (is_logged_in && user_id) {
+                    RudderStack.identifyEvent(user_id, {
+                        language: getLanguage().toLowerCase() || 'en',
+                    });
+                    RudderStack.pageView(current_page);
+                }
+            }
         });
 
         if (scroll_ref.current) setAppContentsScrollRef(scroll_ref);
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    RudderStack.pageView(current_page);
```

Comment:
We need to track the page view every re-render on the root container `app-contents.jsx`, otherwise when user navigates to a different page it won't be called again.
Inside `pageView` there is already a check to ensure that this event is only called once per page